script.js

```diff
@@ -1,4 +1,154 @@
+  } else if (currentGameMode == "sps game") {
+    myOutputValue = playSPSGame(input);
+  }
+  return myOutputValue;
+};
+
+var playSPSGame = function (userInput) {
+  var comOutput = diceRoll();
+  var userObject = icon(userInput);
+  var comObject = icon(comOutput);
+  var reversedOutput = `reversed ${comOutput}`;
+  var myOutputValue = `Input is invalid. Do only enter scissors, paper or stone.`;
+
+  if (userInput == comOutput || userInput == `reversed${comOutput}`) {
+    numberOfDraws += 1;
+    totalNumberOfMatch += 1;
+    userWinningPercentage = getUserWinningPercentage();
+    comWinningPercentage = getComWinningPercentage();
+    myOutputValue = `${userName} chose ${userInput}${userObject} <br> Computer chose ${comOutput}${comObject} <br><br>
+    It's a draw! <br><br>
+    ${userName}'s current score: ${userCurrentScore} out of ${totalNumberOfMatch} (Winning Percentage: ${userWinningPercentage} <br>
+    Computer's current score: ${comCurrentScore} out of ${totalNumberOfMatch} (Winning Percentage: ${comWinningPercentage} <br>
+    Number of draws: ${numberOfDraws}`;
+  }
+
+  if (
+    (userInput == "scissors" && comOutput == "paper") ||
+    (userInput == "paper" && comOutput == "stone") ||
+    (userInput == "stone" && comOutput == "scissors")
+  ) {
+    totalNumberOfMatch += 1;
+    userCurrentScore += 1;
+    userWinningPercentage = getUserWinningPercentage();
+    comWinningPercentage = getComWinningPercentage();
+    myOutputValue = `${userName} chose ${userInput}${userObject} <br>
+    Computer chose ${comOutput}${comObject}<br><br>
+    ${userName} wins! <br><br>
+    ${userName}'s current score: ${userCurrentScore} out of ${totalNumberOfMatch} (Winning Percentage: ${userWinningPercentage} <br>
+    Computer's current score: ${comCurrentScore} out of ${totalNumberOfMatch} (Winning Percentage: ${comWinningPercentage}<br>
+    Number of draws: ${numberOfDraws}`;
+  }
+
+  if (
+    (userInput == "paper" && comOutput == "scissors") ||
+    (userInput == "stone" && comOutput == "paper") ||
+    (userInput == "scissors" && comOutput == "stone")
+  ) {
+    totalNumberOfMatch += 1;
+    comCurrentScore += 1;
+    userWinningPercentage = getUserWinningPercentage();
+    comWinningPercentage = getComWinningPercentage();
+    myOutputValue = `${userName} chose ${userInput}${userObject} <br>
+    Computer chose ${comOutput}${comObject}<br><br>
+    Computer wins! <br><br>
+    ${userName}'s current score: ${userCurrentScore} out of ${totalNumberOfMatch} (Winning Percentage: ${userWinningPercentage}<br>
+    Computer's current score: ${comCurrentScore} out of ${totalNumberOfMatch} (Winning Percentage: ${comWinningPercentage}<br>
+    Number of draws: ${numberOfDraws}`;
+  }
+
+  if (
+    (userInput == "reversed scissors" && comOutput == "paper") ||
+    (userInput == "reversed paper" && comOutput == "stone") ||
+    (userInput == "reversed stone" && comOutput == "scissors")
```

Comment:
You might want to combine the reverse statements with the statements of the normal game that result in the same outcome (user wins/loses/draws) since the code block that runs in both cases are the same